### DISCUSSION TITLE
Question Regarding SC 3.3.2 - Labels or Instructions

### DISCUSSION THREAD

**Discussion by AccessibilityEnthusiastic** [w3c/wcag#3470] [Q&A]
I have a question regarding Success Criterion 3.3.2 - Labels or Instructions in the context of a specific implementation.

The visible label for a "Select All" control is provided using the `<title>` attribute. However, the `HTML <title>` attribute tooltip does not display when the control is in keyboard focus. This means that keyboard users may not have access to the visible label for this control.

On the other hand, the title attribute tooltip is available for mouse users, providing them with information about the visible label.

My question is whether this implementation complies with Success Criterion 3.3.2 or if it raises concerns. Specifically, does the fact that the visible label cannot be accessed by keyboard users due to the reliance on the title attribute mean that it does not meet the requirements of SC 3.3.2?
![Title](https://github.com/w3c/wcag/assets/115781693/32efe992-93ca-4205-a086-149ec5c11961)

**Comment by stevefaulkner** [comment]
> does the fact that the visible label cannot be accessed by keyboard users due to the reliance on the title attribute mean that it does not meet the requirements of SC 3.3.2?

yes

The visible label must be accessible to all users, including keyboard users

**Reply by AccessibilityEnthusiastic** [reply]
title attribute is not keyboard accessible by default, which is consistent with its intended behavior.

My query pertains to whether the lack of keyboard accessibility in the title attribute should lead to a failure of SC 3.3.2 or SC 2.1.1 checkpoints?

There is an exception under SC 1.4.13 (Content on Hover or Focus), which states: "The visual presentation of the additional content is controlled by the user agent and is not modified by the author." This exception specifically references browser tooltips created through the use of the HTML title attribute as an example of additional content controlled by the user agent.

Given this exception, I would like to explore the possibility of applying a similar exception to SC 3.3.2 and SC 2.1.1 checkpoints when the title attribute is used. Since the lack of keyboard accessibility in the title attribute is a known and expected behavior, it may be argued that this feature falls under the same category as browser tooltips mentioned in the exception under SC 1.4.13.
https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html

**Reply by patrickhlauke** [reply]
> My query pertains to whether the lack of keyboard accessibility in the title attribute should lead to a failure of SC 3.3.2 or SC 2.1.1 checkpoints?

Fails both.

The mention in 1.4.13 about something being "controlled by the user agent" applies to 1.4.13, and does not influence pass/fail on other SCs

**Comment by AccessibilityEnthusiastic** [comment]
In these WCAG examples also, the title is used as the visible label for individual fields.
https://www.w3.org/WAI/WCAG22/working-examples/aria-grouping-multipart-fields-ssn/
https://www.w3.org/WAI/WCAG21/working-examples/aria-grouping-related-fields/

**Reply by stevefaulkner** [reply]
The examples both have a permanently visible text label
